.concourse/pipelines/master-pull-request.yaml

```diff
@@ -0,0 +1,182 @@
+                  tag: debian-1.45.0
+                  aws_access_key_id: ((polymath-ecr.aws-access-id))
+                  aws_secret_access_key: ((polymath-ecr.aws-access-secret))
+                  aws_region: us-east-1
+              run:
+                dir: githubpr-polymesh
+                path: bash
+                args: [".concourse/scripts/t-rustfmt-lint.sh"]
+          - task: cargo-build
+            config:
+              inputs:
+                - name: githubpr-polymesh
+                - name: semver-git-polymesh
+              caches:
+                - path: githubpr-polymesh/target
+                - path: /usr/local/cargo
+              platform: linux
+              image_resource:
+                type: registry-image
+                source:
+                  repository: polymathnetwork/rust
+                  tag: debian-1.45.0
+                  aws_access_key_id: ((polymath-ecr.aws-access-id))
+                  aws_secret_access_key: ((polymath-ecr.aws-access-secret))
+                  aws_region: us-east-1
+              run:
+                path: bash
+                args: ["githubpr-polymesh/.concourse/scripts/t-cargo-build.sh", "githubpr-polymesh", "build-caches", "artifact", "semver-git-polymesh"]
+              outputs:
+                - name: build-caches
+                - name: artifact
+          - task: npm-install
+            config:
+              inputs:
+                - name: githubpr-polymesh
+              caches:
+                - path: githubpr-polymesh/scripts/cli/node_modules
+              platform: linux
+              image_resource:
+                type: registry-image
+                source:
+                  repository: polymathnetwork/node
+                  tag: 12
+                  aws_access_key_id: ((polymath-ecr.aws-access-id))
+                  aws_secret_access_key: ((polymath-ecr.aws-access-secret))
+                  aws_region: us-east-1
+              run:
+                path: bash
+                args: ["githubpr-polymesh/.concourse/scripts/t-npm-install.sh", "githubpr-polymesh", "npm-caches"]
+              outputs:
+                - name: npm-caches
+      - in_parallel:
+          - task: cargo-test
+            config:
+              inputs:
+                - name: githubpr-polymesh
+                - name: build-caches
+              caches:
+                - path: githubpr-polymesh/target
+              platform: linux
+              image_resource:
+                type: registry-image
+                source:
+                  repository: polymathnetwork/rust
+                  tag: debian-1.44.1
+                  aws_access_key_id: ((polymath-ecr.aws-access-id))
+                  aws_secret_access_key: ((polymath-ecr.aws-access-secret))
+                  aws_region: us-east-1
+              run:
+                path: bash
+                args: ["githubpr-polymesh/.concourse/scripts/t-cargo-test.sh", "githubpr-polymesh", "build-caches"]
```

Comment:
test cache should ideally be different from build cache. test compiles the binaries in debug mode while the build pipeline uses release mode. This means that the cache generated by these pipelines are not independent of each other and there's no need to increase the size by combining them.